library/core/resource-loader/src/main/java/org/quiltmc/qsl/resource/loader/mixin/client/GameOptionsMixin.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.resource.loader.mixin.client;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.gson.Gson;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.client.option.GameOptions;
+import net.minecraft.resource.ResourcePackManager;
+
+import org.quiltmc.qsl.resource.loader.impl.ModNioResourcePack;
+import org.quiltmc.qsl.resource.loader.impl.ModResourcePackProvider;
+
+@Mixin(GameOptions.class)
+public abstract class GameOptionsMixin {
+	@Shadow
+	public List<String> resourcePacks;
+
+	@Shadow
+	private static List<String> parseList(String content) {
+		throw new UnsupportedOperationException("Injection failed.");
```

Comment:
I think we want to use assertion error in these cases? Unless there is a more appropriate type than that.

Reply:
To me a `IllegalStateException` would be more appropriate:

> Signals that a method has been invoked at an illegal or inappropriate time. In other words, the Java environment or Java application is not in an appropriate state for the requested operation.

Reply:
Yeah that makes sense, we probably need a document noting these little conventions that aren't strictly within scope of code style